includes/rest-api/class-wp-job-manager-models-settings.php

```diff
@@ -38,7 +38,16 @@ public function get_settings() {
 			include_once $parent . '/admin/class-wp-job-manager-settings.php';
 		}
 
-		return WP_Job_Manager_Settings::instance()->get_settings();
+		$settings = WP_Job_Manager_Settings::instance()->get_settings();
+		$filtered_settings = array();
+		foreach ( $settings as $key => $value ) {
```

Comment:
Are you thinking this would be more explicit than `unset( $settings['rest_api'] );`?